application/tests/unit/common/models/MethodTest.php

```diff
@@ -57,5 +60,133 @@ public function testGetMaskedValueEmail()
         $this->assertEquals('e**************9@d*****.o**', $method->getMaskedValue());
     }
 
+    public function testRuleValidateValueAsPhone()
+    {
+        $method = $this->methods('method1');
+        $this->assertTrue($method->validate(['value']));
+
+        $method->value = 'email@domain.com';
+        $this->assertFalse($method->validate(['value']));
+    }
+
+    public function testCreateAndSendVerificationEmail()
+    {
+        EmailUtils::removeEmailFiles();
+        $user = $this->users('user1');
+
+        $this->assertEquals(0, EmailUtils::getEmailFilesCount());
+
+        $method = Method::createAndSendVerification(
+            $user->id,
+            Method::TYPE_EMAIL,
+            'unique-1461443608@email.com'
+        );
+
+        $this->assertEquals(1, EmailUtils::getEmailFilesCount());
+        $this->assertTrue(EmailUtils::hasEmailFileBeenCreated($method->verification_code));
+        $this->assertTrue(EmailUtils::hasEmailFileBeenCreated($method->value));
+
+    }
+
+    public function testCreateAndSendVerificationPhone()
+    {
+        $user = $this->users('user1');
+        $mockPhones = include __DIR__ . '/../../../mock/phone/data.php';
+        $method = Method::createAndSendVerification(
+            $user->id,
+            Method::TYPE_PHONE,
+            $mockPhones[0]['number']
+        );
+
+        $this->assertEquals($mockPhones[0]['code'], $method->verification_code);
+    }
```

Comment:
I may just be missing it, but what is it about the `createAndSendVerification` function that this test is proving works as intended?